artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -1097,26 +1102,32 @@ public void run() {
                // record==null here could only mean there is a compactor
                // computing the delete should be done after compacting is done
                if (jrnRecord == null) {
-                  compactor.addCommandUpdate(id, usedFile, updateRecord.getEncodeSize());
+                  if (compactor != null) {
```

Comment:
Save compactor in a local variable to save it to be loaded more then one time

Reply:
compactor is only changed on writeLock(). it is safe to use it here.

Reply:
Yes but it still an unnecessary cost: each time we use it, the jit will load it and null check it implicitly

Reply:
I don't think that's a big deal but I will do it